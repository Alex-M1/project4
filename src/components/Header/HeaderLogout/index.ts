import HeaderLogout from './HeaderLogout';

export default HeaderLogout;
