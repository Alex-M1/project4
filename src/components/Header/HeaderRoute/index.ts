import HeaderRoute from './HeaderRoute';

export default HeaderRoute;
