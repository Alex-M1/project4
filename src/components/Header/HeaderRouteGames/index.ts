import HeaderRouteGames from './HeaderRouteGames';

export default HeaderRouteGames;
