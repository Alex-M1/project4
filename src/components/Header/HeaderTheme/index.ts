import HeaderTheme from './HeaderTheme';

export default HeaderTheme;
