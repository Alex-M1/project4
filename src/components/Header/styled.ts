import { changeTheme } from 'src/helpers/changeTheme';
import styled from 'styled-components';
import { IStyled } from '../_common_/types/styledTypes';

export const StHeader = styled.div<IStyled>`
    display: flex;
    justify-content: space-around;
    align-items: center;
    width: 100vw;
    height: 90px;
    background: ${(props) => changeTheme(props, 'headerBg')}
`;
export const StLogo = styled.img <IStyled>`
  display: block;
  cursor: pointer;
  filter: drop-shadow(0 0 2px ${(props) => changeTheme(props, 'headerBg')});
`;
export const StHeaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const StNavContainer = styled.div`
  width: 150px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
