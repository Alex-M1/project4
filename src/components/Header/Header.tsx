import React, { useContext, useMemo } from 'react';

import { StHeader, StLogo, StHeaderContainer, StNavContainer } from './styled';
import { Theme } from '../hocs';
import HeaderTheme from './HeaderTheme';
import HeaderLanguage from './HeaderLanguage';
import HeaderRoute from './HeaderRoute';
import HeaderLogout from './HeaderLogout';

const Header: React.FC = ({ location }) => {
    const { theme, changeTheme } = useContext(Theme);
    const toggleThemeMode = () => {
        changeTheme(theme === 'dark' ? 'light' : 'dark');
      };
    const isNav = useMemo(() => {
        const path = location.pathname;
        if (path === '/' || path === '/signup') {
            return null;
        }
        return <HeaderRoute/>;
    }, [location.pathname]);

    return (
        <StHeader>
            <StHeaderContainer>
                <StLogo
                    src="src/assets/img/logo.png"
                    alt="logo"
                />
            </StHeaderContainer>
            <StHeaderContainer>
                {isNav}
            </StHeaderContainer>
            <StHeaderContainer>
                <StNavContainer>
                    <HeaderTheme
                        toggleThemeMode={toggleThemeMode}
                    />
                    <HeaderLanguage/>
                    <HeaderLogout/>
                </StNavContainer>
            </StHeaderContainer>
        </StHeader>
    );
};

export default Header;
