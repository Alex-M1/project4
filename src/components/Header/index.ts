import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { leaveRoom } from 'store/room/actions';
import Header from './Header';

const mapDispatchToProps = {
  leaveRoom,
};

export default connect(null, mapDispatchToProps)(withRouter(Header));
