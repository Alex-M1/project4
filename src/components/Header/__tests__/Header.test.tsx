import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);

describe('Header', () => {
  it('Should match snapshot', () => {
    const component = shallow(<Header />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StHeader', () => {
    const component = shallow(<Header />);
    expect(component.find('styled__StHeader')).toHaveLength(1);
  });
});
