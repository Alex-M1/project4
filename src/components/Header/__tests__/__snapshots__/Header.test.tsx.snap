// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<div class=\\"styled__StHeader-hIeMMX kAxmnQ\\"><div class=\\"styled__StHeaderContainer-bIRgow hYa-DCe\\"><div class=\\"styled__StNavContainer-jDboJm kGeIWZ\\"><img src=\\"src/assets/img/logo.png\\" alt=\\"logo\\" class=\\"styled__StLogo-gRpOwf bEQhLK\\"/></div></div><div class=\\"styled__StHeaderContainer-bIRgow hYa-DCe\\"><div class=\\"styled__StWrapper-kHlHLr fkLuEu\\"><a href=\\"/main\\"><p class=\\"styled__StP-gOoQqg ktsAUY\\"></p></a><a href=\\"/statistic\\"><p class=\\"styled__StP-gOoQqg ktsAUY\\"></p></a></div></div><div class=\\"styled__StHeaderContainer-bIRgow hYa-DCe\\"><div class=\\"styled__StNavContainer-jDboJm kGeIWZ\\"><button class=\\"styled__StThemeToggle-jPlXPr hdfHcR\\"></button><button class=\\"styled__StLangToggle-gXeXkO bpAlfl\\">En</button><a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\"><img src=\\"src/assets/img/logout.svg\\" class=\\"styled__StDoor-kiqXlu ixKzTX\\"/></a></div></div></div> "`;
